src/fi_tostr.c

```diff
@@ -72,15 +72,15 @@
 
 #define OFI_BUFSIZ 8192
 
-static void ofi_tostr_fid(const char *label, char *buf, const struct fid *fid)
+static void ofi_tostr_fid(const char *label, char *buf, size_t buflen, const struct fid *fid)
 {
 	if (!fid || !FI_CHECK_OP(fid->ops, struct fi_ops, tostr))
-		ofi_strcatf(buf, "%s%p\n", label, fid);
+		ofi_strcatf(buf, buflen, "%s%p\n", label, fid);
 	else
-		fid->ops->tostr(fid, buf, OFI_BUFSIZ - strnlen(buf, OFI_BUFSIZ));
+		fid->ops->tostr(fid, buf, buflen - strnlen(buf, buflen));
 }
 
-static void ofi_tostr_opflags(char *buf, uint64_t flags)
+static void ofi_tostr_opflags(char *buf, size_t buflen, uint64_t flags)
```

Comment:
This function isn't using buflen, and probably should be now.  Maybe IFFLAGSTR needs to be updated?

Reply:
`IFFLAGSTR` implicitly uses `buf` and `buflen`, even though those variables aren't given as arguments to the macro. Here is how IFFLAGSTR is declared:
```
 #define IFFLAGSTR(flags, SYM) \
       do { if (flags & SYM) ofi_strncatf(buf, buflen, #SYM ", "); } while(0)
```
The implicit argument `buf` was already there before my patch, so I just followed the coding style of the existing code when I added `buflen`. If you prefer I can update IFFLAGSTR to pass `buf` and `buflen` as arguments.